### PR TITLE
*: Support Mod(N, M)

### PR DESCRIPTION
**Pull request by shenli**
Fix: https://github.com/pingcap/tidb/issues/1193

See: http://dev.mysql.com/doc/refman/5.7/en/mathematical-functions.html#function_mod

### CODE REVIEW

File: evaluator/evaluator_test.go [hunk 1]
```diff
@@ -998,3 +998,21 @@ func (s *testEvaluatorSuite) TestEvaluatedFlag(c *C) {
...
+			resultStr: "4",
+		},
+		{
+			exprStr:   "MOD(29, 9)",
+			resultStr: "2",
+		},
+		{
+			exprStr:   "MOD(34.5, 3)",
+			resultStr: "1.5",
+		},
```
Comment (zimulala) [inline evaluator/evaluator_test.go]: Add `MOD(N,0) returns NULL.` test ?

**Comment by coocood** [comment]
LGTM

File: evaluator/evaluator_test.go (hunk 1 above)
Comment (shenli) [inline evaluator/evaluator_test.go]: There are already some test cases for this.

**Comment by zimulala** [comment]
LGTM

**Pull request merged**